### PR TITLE
[faker.js] updates for v5

### PR DESCRIPTION
**Pull request by TastefulElk**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/Marak/faker.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by TastefulElk** [comment]
All updates required for Faker v5 release except for `finance` related changes which were added by @wit1312 in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/47283

File: types/faker/faker-tests.ts [hunk 1]
```diff
@@ -182,9 +240,19 @@ resultNum = faker.random.number({
     max: 0,
     precision: 0
 });
+resultNum = faker.random.float();
+resultNum = faker.random.float(0);
+resultNum = faker.random.float({
+	min: 0,
```
Comment (danger-public) [inline types/faker/faker-tests.ts:246]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by danger-public** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  **types/faker/index.d.ts#L360** - The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.
  </td>
    </tr>
  </tbody>
</table>



Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## faker ([<kbd>unpkg</kbd>](https://unpkg.com/browse/faker@latest/))
was missing the following properties:
1. locales
1. localeFallback
1. definitions
1. unique
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a9ea6f792baceba080cdb6a637afcc15a5395631
</p>

**Review by mastermatt — APPROVED** [review]

**Comment by TastefulElk** [comment]
@bensw @basp @Kuniwak @mattbishop @testica  any of you wanna take a look?

**Comment by TastefulElk** [comment]
@bensw @basp @Kuniwak @mattbishop @testica sorry to bug you guys again but could any of you please have a look at this so we can fix the typings for the new version 5?

**Review by bensw — APPROVED** [review]
Sorry for the delay. This LGTM. Thank you for your hard work.


(Also it has been a few years since I've contributed to this repo, so let me know if there's any other action I need to take other than approving it to move it forward).

**Comment by TastefulElk** [comment]
Ready to merge

**Pull request merged**